scripts/histogram_test.jl

```diff
@@ -0,0 +1,150 @@
+        ∘ ConstValueFunctions.filter_early_term
+    )
+
+    redshift_vf = (
+        ValueFunction(AccretionFormulae.redshift)
+        ∘ FilterValueFunction((m, sol, max_time; kwargs...) -> sol.u[end][2] > R_isco, NaN)
+        ∘ ConstValueFunctions.filter_early_term   
+    )
+
+    # do the render
+    temperature_img = @time rendergeodesics(
+        m, u, 2000.0, 
+        d, 
+        fov_factor=fov*size_multiplier, abstol=tolerance, reltol=tolerance,
+        image_width = 350*size_multiplier,
+        image_height = 250*size_multiplier,
+        vf = temperature_vf,
+        dtmax = dtmax
+    )
+
+    radius_img = @time rendergeodesics(
+        m, u, 2000.0, 
+        d, 
+        fov_factor=fov*size_multiplier, abstol=tolerance, reltol=tolerance,
+        image_width = 350*size_multiplier,
+        image_height = 250*size_multiplier,
+        vf = radius_vf,
+        dtmax = dtmax
+    )
+
+    redshift_img = @time rendergeodesics(
+        m, u, 2000.0, 
+        d, 
+        fov_factor=fov*size_multiplier, abstol=tolerance, reltol=tolerance,
+        image_width = 350*size_multiplier,
+        image_height = 250*size_multiplier,
+        vf = redshift_vf,
+        dtmax = dtmax
+    )
+    # correcting for physical mass
+    M_phys = mass*1.99e30
+    r_isco_phys = AccretionFormulae.r_isco(M_phys, 0.998)
+    r_g_phys = M_phys
+
+    numerators =   AccretionFormulae.mass_scale_fraction.(M_phys, η_phys, edd_ratio_phys, r_isco_phys, r_g_phys, radius_img*M_phys)
+    denominators = AccretionFormulae.mass_scale_fraction.(M, η, edd_ratio, R_isco, M, radius_img)
+
+    fractions = numerators./denominators
+    temperature_img .*= fractions
+
+    # # autoscale
+    # scale = maximum(filter(!isnan,temperature_img))
+    # scale = floor(log(10, scale))
+    # scale = 10^scale
+    
+    # scaling image
+    scale = 1e7
+    scalestr = @sprintf "%.E" scale
+    new_img = reverse(temperature_img, dims=1)
+    new_img ./= scale
+
+    # hm = heatmap(new_img, aspect_ratio=1.0, size=(resolution*3/2, resolution), 
+    # clim=(0,3)
+    # )
+    # # contour(new_img, aspect_ratio=1.0, size=(resolution*3/2, resolution), clim=(0,3))
+    # title!("Temperature Scale = $scalestr, Mass = $mass M_☼")
+    # # display(hm)
+    return temperature_img, redshift_img
+end
+
+
+function energy_histogram(;obs_angle=85.0)
+    temperature_img, redshift_img = temperature_render(obs_angle=obs_angle)
+
+
+    # constants
+    h = 6.63e-34
+    c = 3e8
+
+    lines=[]
+    for (i, x) in enumerate(temperature_img)
+        for (j, y) in enumerate(x)
+            g = redshift_img[i][j]
+            energy = 5.52e18*h*c*y
+            # @show(energy)
+            push!(lines, energy*g^4)
+        end
+
+    end
+
+    histogram(lines, 
+            xlims=(0,10),
+            nbins=1000)
+    title!("Observation Angle = $obs_angle")
+    # temperature_render(obs_angle=85.0, mass=1)
+end
+
+energy_histogram()
```

Comment:
Post plot too?

Reply:
![plot_3](https://user-images.githubusercontent.com/94124028/158217152-4b492317-2117-496a-94fc-53a377434dc5.png)

This is just a basic one we made, and doesn't really look right, but some of the information [below](https://github.com/astro-group-bristol/AccretionFormulae.jl/pull/25#issuecomment-1066996047) should help us.